src/shared/components/Uploader/Uploader.tsx

```diff
@@ -197,14 +201,51 @@ export const Uploader = ({
 
   const placeholderImgId = isPrimaryUiType ? 'img-filled' : 'img-outlined';
   const placeholderImgSize = isPrimaryUiType ? 32 : 24;
-  const hasSizeError = error === UploadFileError.Size;
-  const hasFormatError = error === UploadFileError.Format;
-  const hasDimensionsError = error === UploadFileError.Dimensions;
+  const [hasSizeError, hasFormatError, hasDimensionsError] = useMemo(
+    () => [
+      error === UploadFileError.Size,
+      error === UploadFileError.Format,
+      error === UploadFileError.Dimensions,
+    ],
+    [error],
+  );
   const hasImageError = hasSizeError || hasFormatError || hasDimensionsError;
   const spinnerUiType = isPrimaryUiType ? SpinnerUiType.Primary : SpinnerUiType.Secondary;
 
   const fileName = imageField?.split('/').at(-1) || image?.name || '';
 
+  useEffect(() => {
```

Comment:
Here, as well let's get rid of `useEffect` that could be replaced with a callback into the places where `error` set to `Size` and `Format` accordingly (in row 110 and 111) , I suppose, when the condition `isPrimaryUiType === false`